File: src/main/scala/com/github/mumoshu/ssp/scala/time/AbstractTime.scala
package com.github.mumoshu.ssp.scala.time

trait AbstractTime {
  var hour: Int
  var minute: Int
}

/*
trait AbstractTime {
  // 抽象var hourは以下のgetter/setterに展開される
  def hout: Int
  def hout_=(x: Int)
  // 抽象var minuteは以下のgetter/setterに展開される
  def minute: Int
  def minute_=(x: Int)
}
*/

File: src/main/scala/com/github/mumoshu/ssp/java/time/AbstractTime.java
package com.github.mumoshu.ssp.java.time;

public interface AbstractTime {
    //var hour: Int
    void setHour(Integer hour); // Scala の Int は Java の int にマッピングされるが、プリミティブではなくクラス扱いなので、Integerとintどっちに例えるか悩む。
    Integer getHour();

    // var minute: Int
    void setMinute(Integer minute);
    Integer getMinute();
}
